Feature/ssl and redirect

berkes opened the pull request:
Fix '*74 no "ssl_certificate" is defined in server' error.

Error caused in unicorn nginx.conf. When ssl was on and site had redirect-domains. 

This also adds ssl-support for redirect-domains in passenger: allowing https versions to be redirected as well.

berkes committed:
Implement custom blocks to be inserted in nginx config.

michiels committed:
Use before_server and after_server for nginx_custom blocks and add sample in Vagrantfile

berkes committed:
Bump version and add note to changelog.

michiels committed:
Correctly check if password attribute is entered for a sysadmin

michiels committed:
Change .ruby-version to 2.2.0

michiels committed:
Some copy changes in CHANGELOG

michiels committed:
Remove commented local dev lines from Cheffile

michiels committed:
Update all depending cookbooks

michiels committed:
Remove instructions to set Ruby version since this is now taken care of by the recipe

michiels committed:
Changelog copy changes

michiels committed:
Added a test kitchen file for using Vagrant

michiels committed:
Renamed default kitchen to .kitchen.digitalocean.yml so you can switch locally

michiels committed:
Revert change in .kitchen.yml because .kitchen.local.yml is for that.

michiels committed:
Do not git ignore .kitchen.local.yml.  You can use this for Vagrant tests

michiels committed:
Make sure that our "vagrant" user automatically gets sudo-et

michiels committed:
Set 2.3.0 release date in Changelog and add Test Kitchen Changelog item

jvanbaarsen committed:
Fixed changelog for 2.4.0 :balloon:

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

michiels committed:
Add rbenv dependency to rails cookbook

oiuzikov committed:
Fixed #165 the `table "*"` option in mysql_database resource is not
valid anymore for database cookbook v2.3.1

oiuzikov committed:
Updated changelog

jvanbaarsen committed:
Changed release date

jvanbaarsen committed:
Downgrade ruby to 2.1.5 for the moment

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Added changelog entry

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Allow redirect domains for passenger stack as well

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Have a correct SSL redirect (user certs)

Fixes #176

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

joshuajansen committed:
Sets encoding to utf-8 for sysadmins metadata

joshuajansen committed:
Adds encoding fix to changelog

jvanbaarsen committed:
Fixed minor typo

jvanbaarsen committed:
Add ruby 2.1.5 ruby version

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Check if enable_ssl is set, otherwise check for certificate

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

hapiben committed:
Removed bin as part of linked directories in Capistrano.

jvanbaarsen committed:
Touch nginx when certificate files are created

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Added the ability to remove your application

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

joshuajansen committed:
Makes nginx: client_max_body_size configurable per app

- Adds client max body size per app config to template
- Set client_max_body size variable for app configs
- Adds change to changelog
- Only set client_max_body_size when valid
- Adds client_max_body_size to sample_host

jvanbaarsen committed:
Added libffi-dev as a system package for ruby 2.2.x instalation

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Add ruby 2.2.1 to the available binaries list

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Add ruby 2.2.1 binary to changelog

pkotnis committed:
Update CONTRIBUTING.md

Incorrectly formatted markdown link.

joshuajansen committed:
Sets secure defaults for SSL protocol and ciphers

joshuajansen committed:
Sets prefer_server_ciphers on and add ciphers

joshuajansen committed:
Sets defaults for ssl protocol for redirect domain

joshuajansen committed:
Disables RC4 cipher

jvanbaarsen committed:
Install ruby 2.1.0 from binaries since source is broken

Ref: https://github.com/sstephenson/ruby-build/issues/526

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Add the unattended-upgrades recipe and set to false by default

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Added unattended upgrades to the sample file

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

jvanbaarsen committed:
Syntax error in the vagrant file

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

michiels committed:
Updated CHANGELOG with release date for 2.4.0

michiels committed:
Use Ruby 2.2.0 as default in Vagrantfile

michiels committed:
Refactor libffi-dev dependency into rails::setup for both Unicorn and Passenger stacks

michiels committed:
Update Vagrantfile to 2.2.0 Ruby for testing

michiels committed:
Update release date to 2015-04-28

michiels committed:
Add unattended-upgrades recipe for regular unicorn rails stack

michiels committed:
Added a changelog entry for Unattended Upgrades

michiels committed:
Remove the default attribute for disabling unattended upgrades, this is already the default in the apt cookbook

michiels committed:
Make changelog entry for unattended upgrades clear

jvanbaarsen committed:
Allow for setting DB pool size via ENV vars

In some cases the default pool size of 5 might be to little, by allowing you to
edit the ENV vars of the app, you can easily change this var.

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

berkes committed:
Fix '*74 no "ssl_certificate" is defined in server' errors

Nginx cannot listen on ssl when there are no certificates

berkes committed:
Allow passenger version to redirect ssl pages too.

jvanbaarsen commented:
@berkes Thanks! Can you also add an entry to the CHANGELOG?

berkes committed:
Add entries to CHANGELOG

berkes commented:
Added entries to CHANGELOG. Had to rebase and force-push for that (sorry).

jvanbaarsen commented:
No problem :-) I'll be on vacation next week, will test this when I get back on Tuesday!

berkes committed:
Bump version number of rails recipe

yourivdlans committed:
Fix box could not be found

Renamed chef to bento to get correct box.

berkes commented:
Anything needed from me to take this forward?

jvanbaarsen commented:
@berkes I don't think anything from your side is required. Its mostly a massive time constraint on my end. Maybe @michiels can also look at this PR so we can speed things up a little?

jvanbaarsen commented on CHANGELOG.md:
```diff
@@ -6,6 +6,8 @@ This project makes use of the [Sementic Versioning](http://semver.org/)
 
 ### Added
 - Set the max db `pool` size via an ENV var called `DB_POOL_SIZE`
+- Fixed bug when redirect-domains were set on a site with SSL.
```
Can you put this under a heading called `### Fixed`? (That way its easier for people that ran into this issue to spot if an issue has been solved for them :) )

jvanbaarsen commented:
Looks good to me. @michiels Any thoughts?

michiels commented:
Looks good! Happy to merge this in to also support the SSL for the redirect domains :)

michiels commented:
Ah, only Jeroen's changelog comment to correct and then let's push the merge button!

michiels committed:
Bump Knife solo gem and Chef to version 12

berkes committed:
Add  client_max_body_size directive to the nginx conf used in unicorn setup

This in addition to 4d4a710a270cb982e3601a362f038278a8f1fb78 where it
was added to the nginx conf in the passenger setup.

berkes committed:
Bump version.

berkes committed:
Add client_max_body_size change to CHANGELOG

berkes committed:
Merge branch 'master' into develop

* master: (61 commits)
  Add client_max_body_size change to CHANGELOG
  Bump version.
  Add  client_max_body_size directive to the nginx conf used in unicorn setup
  Bump Knife solo gem and Chef to version 12
  Fix box could not be found
  Allow for setting DB pool size via ENV vars
  Make changelog entry for unattended upgrades clear
  Remove the default attribute for disabling unattended upgrades, this is already the default in the apt cookbook
  Added a changelog entry for Unattended Upgrades
  Add unattended-upgrades recipe for regular unicorn rails stack
  Update release date to 2015-04-28
  Update Vagrantfile to 2.2.0 Ruby for testing
  Refactor libffi-dev dependency into rails::setup for both Unicorn and Passenger stacks
  Use Ruby 2.2.0 as default in Vagrantfile
  Updated CHANGELOG with release date for 2.4.0
  Syntax error in the vagrant file
  Added unattended upgrades to the sample file
  Add the unattended-upgrades recipe and set to false by default
  Install ruby 2.1.0 from binaries since source is broken
  Disables RC4 cipher
  ...
